src/search.c

```diff
@@ -178,10 +178,21 @@ rs_search_for_block(rs_weak_sum_t weak_sum,
     int l = bucket->l; /* left bound of search region */
     int r = bucket->r + 1; /* right bound of search region */
     int v = 1; /* direction of next move: -ve left, +ve right */
+    int i = -1;
 
     if (l == NULL_TAG)
         return 0;
 
+    /* Try a direct match */
+    if ((job_token > 0) && (job_token < sig->count) &&
+        (weak_sum == sig->block_sigs[job_token-1].weak_sum)) {
```

Comment:
Fair enough. In the new hashtable I omit duplicate blocks, b/c only the first one can ever be matched.